runtime/runtime/src/lib.rs

```diff
@@ -1335,6 +1335,30 @@ impl Runtime {
 
         let (trie_changes, state_changes) = state_update.finalize()?;
 
+        // Precompile contracts just inserted into state and store result (compiled code or error)
+        // in the database.
+        // Note that contract compilation costs are already accounted in deploy cost using
+        // special logic in estimator (see get_runtime_config() function).
+        let wasm_config = apply_state.config.wasm_config.clone();
+        let cache = apply_state.cache.as_deref();
+        state_changes.par_iter().for_each(|state_change| match state_change.trie_key {
+            TrieKey::ContractCode { .. } => {
+                let code = state_change
+                    .changes
+                    .last()
+                    .expect("Committed entry should have at least one change")
+                    .data
+                    .as_ref();
+                // If code is None, it means that it was just removed from the state, so there is
+                // nothing to precompile.
+                if let Some(code) = code {
+                    let contract_code = ContractCode::new(code.clone(), None);
+                    precompile_contract(&contract_code, &wasm_config, cache).ok();
+                }
+            }
+            _ => {}
+        });
```

Comment:
We already compile contracts when they are deployed. You don't need to recompile them again here